### PR TITLE
Added player.isDucking()

### PR DESCRIPTION
**Pull request by thecatontheflat**
I couldn't find "How to contribute" guide in the Readme. Could you please guide me through the process, so I could update the dist and the docs? I've tried `npm run build` and `npm run generate-docs`, but the build was a mess of unexpected diffs (e.g. spaces changed from 4 to 2).

### CODE REVIEW

**Comment by saul** [comment]
Thanks for your changes. I've added some contributing guidelines to the repo now. Prettier runs on commit, so the code style will be normalised at commit time.

As for updating the docs etc, I'll do that when I release a new version.

**Comment by saul** [comment]
IIRC `m_bDucking` is only set when the player is transitioning from standing -> ducked. Once they are ducked, `m_bDucked` is set instead. Are you sure this is the prop you mean?

**Comment by master117** [comment]
I'm pretty sure @saul is right. But haven't stepped through it in a while. An easy test case/proof of concept would be to look at the average ducking duration for a match with this. Or record and time a single duck.

**Comment by saul** [comment]
Judging from the source code, this is indeed the case: https://github.com/ValveSoftware/source-sdk-2013/blob/master/mp/src/game/shared/gamemovement.cpp

**Comment by thecatontheflat** [comment]
Nice! Thank you for tipping about `m_bDucking` and `m_bDucked`, this is indeed what I was looking for. I will add another method to the PR too!

File: src/entities/player.ts [hunk 1]
```diff
@@ -429,6 +429,20 @@ export class Player extends BaseEntity<CCSPlayer> {
   get isWalking(): boolean {
     return this.getProp("DT_CSPlayer", "m_bIsWalking");
   }
+    
+  /**
+   * @returns Is ducking
```
Comment (saul) [inline src/entities/player.ts]:
```suggestion
   * @returns Player is transitioning from ducked -> standing or standing -> ducked
```

**Comment by saul** [comment]
Thanks for this - will get a new version rolled out today

**Pull request merged**